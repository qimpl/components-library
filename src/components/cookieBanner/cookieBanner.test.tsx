import React from 'react';

import { render } from '@testing-library/react';

import CookieBanner from 'components/cookieBanner';

test('Simple Cookie Banner', () => {
  expect(
    render(
      <CookieBanner
        ariaAcceptLabel='Ok'
        buttonText='Ok'
        message='Lorem ipsum dolor sit amet, consectetur adipisicing elit. Fuga eius quisquam dignissimos vitae ducimus atque, est ipsa vero rem veniam quam. Ea tenetur odio quaerat nulla autem voluptate quo maiores.'
        titleContent='We love cookies'
      />
    )
  ).toMatchSnapshot();
});

test('Cookie Banner with element title', () => {
  expect(
    render(
      <CookieBanner
        ariaAcceptLabel='Ok'
        buttonText='Ok'
        message='Lorem ipsum dolor sit amet, consectetur adipisicing elit. Fuga eius quisquam dignissimos vitae ducimus atque, est ipsa vero rem veniam quam. Ea tenetur odio quaerat nulla autem voluptate quo maiores.'
        titleContent={
          <>
            We love cookies{' '}
            <span role='img' aria-label='cookie'>
              🍪
            </span>
          </>
        }
      />
    )
  ).toMatchSnapshot();
});
