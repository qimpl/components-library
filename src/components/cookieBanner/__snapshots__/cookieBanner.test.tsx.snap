// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie Banner with element title 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CookieConsent"
        style="align-items: center; background: rgb(53, 53, 53); color: white; display: flex; flex-wrap: wrap; justify-content: center; left: 0px; position: fixed; width: auto; z-index: 999; margin: 10px; padding: 20px; background-color: white; border-radius: 22px; box-shadow: 0 0 19px 3px rgba(0, 0, 0, 0.2); flex-direction: row; bottom: 0px;"
      >
        <div
          class=""
          style="flex: 1 0 300px; margin: 15px; color: rgb(1, 60, 99);"
        >
          <h3
            class="sc-hKgILt eXHKDY"
            font-size="16px"
          >
            We love cookies
             
            <span
              aria-label="cookie"
              role="img"
            >
              üç™
            </span>
          </h3>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Fuga eius quisquam dignissimos vitae ducimus atque, est ipsa vero rem veniam quam. Ea tenetur odio quaerat nulla autem voluptate quo maiores.
        </div>
        <div
          class=""
        >
          <button
            aria-label="Ok"
            class=""
            id="rcc-confirm-button"
            style="background: rgb(255, 212, 45); border: 0px; border-radius: 50px; box-shadow: none; color: white; cursor: pointer; flex: 0 0 auto; padding: 13px 10px; margin: 15px; width: 150px; height: 40px; background-color: rgb(1, 60, 99);"
          >
            Ok
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CookieConsent"
      style="align-items: center; background: rgb(53, 53, 53); color: white; display: flex; flex-wrap: wrap; justify-content: center; left: 0px; position: fixed; width: auto; z-index: 999; margin: 10px; padding: 20px; background-color: white; border-radius: 22px; box-shadow: 0 0 19px 3px rgba(0, 0, 0, 0.2); flex-direction: row; bottom: 0px;"
    >
      <div
        class=""
        style="flex: 1 0 300px; margin: 15px; color: rgb(1, 60, 99);"
      >
        <h3
          class="sc-hKgILt eXHKDY"
          font-size="16px"
        >
          We love cookies
           
          <span
            aria-label="cookie"
            role="img"
          >
            üç™
          </span>
        </h3>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Fuga eius quisquam dignissimos vitae ducimus atque, est ipsa vero rem veniam quam. Ea tenetur odio quaerat nulla autem voluptate quo maiores.
      </div>
      <div
        class=""
      >
        <button
          aria-label="Ok"
          class=""
          id="rcc-confirm-button"
          style="background: rgb(255, 212, 45); border: 0px; border-radius: 50px; box-shadow: none; color: white; cursor: pointer; flex: 0 0 auto; padding: 13px 10px; margin: 15px; width: 150px; height: 40px; background-color: rgb(1, 60, 99);"
        >
          Ok
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Simple Cookie Banner 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CookieConsent"
        style="align-items: center; background: rgb(53, 53, 53); color: white; display: flex; flex-wrap: wrap; justify-content: center; left: 0px; position: fixed; width: auto; z-index: 999; margin: 10px; padding: 20px; background-color: white; border-radius: 22px; box-shadow: 0 0 19px 3px rgba(0, 0, 0, 0.2); flex-direction: row; bottom: 0px;"
      >
        <div
          class=""
          style="flex: 1 0 300px; margin: 15px; color: rgb(1, 60, 99);"
        >
          <h3
            class="sc-hKgILt eXHKDY"
            font-size="16px"
          >
            We love cookies
          </h3>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Fuga eius quisquam dignissimos vitae ducimus atque, est ipsa vero rem veniam quam. Ea tenetur odio quaerat nulla autem voluptate quo maiores.
        </div>
        <div
          class=""
        >
          <button
            aria-label="Ok"
            class=""
            id="rcc-confirm-button"
            style="background: rgb(255, 212, 45); border: 0px; border-radius: 50px; box-shadow: none; color: white; cursor: pointer; flex: 0 0 auto; padding: 13px 10px; margin: 15px; width: 150px; height: 40px; background-color: rgb(1, 60, 99);"
          >
            Ok
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CookieConsent"
      style="align-items: center; background: rgb(53, 53, 53); color: white; display: flex; flex-wrap: wrap; justify-content: center; left: 0px; position: fixed; width: auto; z-index: 999; margin: 10px; padding: 20px; background-color: white; border-radius: 22px; box-shadow: 0 0 19px 3px rgba(0, 0, 0, 0.2); flex-direction: row; bottom: 0px;"
    >
      <div
        class=""
        style="flex: 1 0 300px; margin: 15px; color: rgb(1, 60, 99);"
      >
        <h3
          class="sc-hKgILt eXHKDY"
          font-size="16px"
        >
          We love cookies
        </h3>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Fuga eius quisquam dignissimos vitae ducimus atque, est ipsa vero rem veniam quam. Ea tenetur odio quaerat nulla autem voluptate quo maiores.
      </div>
      <div
        class=""
      >
        <button
          aria-label="Ok"
          class=""
          id="rcc-confirm-button"
          style="background: rgb(255, 212, 45); border: 0px; border-radius: 50px; box-shadow: none; color: white; cursor: pointer; flex: 0 0 auto; padding: 13px 10px; margin: 15px; width: 150px; height: 40px; background-color: rgb(1, 60, 99);"
        >
          Ok
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
